File: backend/Domain/Dtos/Responses/TokenResponse.cs
using System.Text.Json.Serialization;

namespace Domain.Dtos.Responses;

public class TokenResponse<TResponse> : Response
{
	public TResponse Data { get; set; }
	public AuthToken AuthToken { get; set; }
	public RefreshToken RefreshToken { get; set; }

	public static TokenResponse<TResponse> Success(TResponse response, AuthToken authToken, RefreshToken refreshToken) =>
		new()
        {
			IsSuccessful = true,
			Data = response,
			AuthToken = authToken,
			RefreshToken = refreshToken
		};
		
	public static TokenResponse<TResponse> Success(AuthToken authToken, RefreshToken refreshToken) =>
		new()
        {
			IsSuccessful = true,
			AuthToken = authToken,
			RefreshToken = refreshToken
		};
}

File: frontend/src/app/models/responses/token-response.ts
import { AuthToken } from "../auth/auth-token";
import { RefreshToken } from "../auth/refresh-token";
import { Response } from "../base/response";

export class TokenResponse<TEntity> extends Response {
	Data: TEntity;
	AuthToken: AuthToken;
	RefreshToken: RefreshToken;
}
